sklearn/utils/_metadata_requests.py

```diff
@@ -1084,6 +1237,11 @@ class attributes, as well as determining request keys from method
             method = attr[attr.index(substr) + len(substr) :]
             for prop, alias in value.items():
                 getattr(requests, method).add_request(param=prop, alias=alias)
+
+        # this indicates that the user has not set any request values for this
+        # object
+        requests._is_default_request = True
```

Comment:
Can this be set closer to the top of this method?

Reply:
No. `add_request` sets it to `False`. So we only set this after having set all the default values.